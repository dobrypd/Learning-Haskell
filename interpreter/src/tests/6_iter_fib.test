int fib(int n) {
    if(n == 0) return 0;
    int a, b, tmp;
    a = 0; b = 1;
    int i =0;
    while(i < n-1) {
        tmp = a;
        a = b;
        b = tmp;
        b = b+a;
        i = i+1;
    };
    return b;
}

int main() {
    return fib(300);
}
