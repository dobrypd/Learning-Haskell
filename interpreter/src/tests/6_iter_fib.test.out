Welcome to Kappa interpreter
222232244629420445529739893461909967206666939096499764990979600
[(fromList [],fromList [(Ident "fib",([Ident "n"],[SStm (IfS (SIf (Eeq (Evar [Ident "n"]) (Econst (Eint 0))) (JumpS (SjumpFive (Econst (Eint 0)))))),SStm (DecS (Ddec (Declarators Type_specifier_int [OnlyDecl (VarName (Ident "a")),OnlyDecl (VarName (Ident "b")),OnlyDecl (VarName (Ident "tmp"))]))),SStm (ExprS (Sexpr (Eassign [Ident "a"] (Econst (Eint 0))))),SStm (ExprS (Sexpr (Eassign [Ident "b"] (Econst (Eint 1))))),SStm (DecS (Ddec (Declarators Type_specifier_int [InitDecl (VarName (Ident "i")) (InitExpr (Econst (Eint 0)))]))),SStm (IterS (SiterWhile (Elthen (Evar [Ident "i"]) (Eminus (Evar [Ident "n"]) (Econst (Eint 1)))) (ListS [SStm (ExprS (Sexpr (Eassign [Ident "tmp"] (Evar [Ident "a"])))),SStm (ExprS (Sexpr (Eassign [Ident "a"] (Evar [Ident "b"])))),SStm (ExprS (Sexpr (Eassign [Ident "b"] (Evar [Ident "tmp"])))),SStm (ExprS (Sexpr (Eassign [Ident "b"] (Eplus (Evar [Ident "b"]) (Evar [Ident "a"]))))),SStm (ExprS (Sexpr (Eassign [Ident "i"] (Eplus (Evar [Ident "i"]) (Econst (Eint 1))))))]))),SStm (DecS Dempty),SStm (JumpS (SjumpFive (Evar [Ident "b"])))])),(Ident "main",([],[SStm (JumpS (SjumpFive (Efunk (Ident "fib") [Econst (Eint 300)])))]))],fromList [])]
